EssentialsDiscord/src/main/java/net/essentialsx/discord/util/DiscordUtil.java

```diff
@@ -165,4 +170,23 @@ public static boolean hasRoles(Member member, List<String> roleDefinitions) {
         }
         return false;
     }
+
+    public static void dispatchDiscordMessage(final JDADiscordService jda, final MessageType messageType, final String message, final boolean allowPing, final String avatarUrl, final String name, final UUID uuid) {
+        if (jda.getPlugin().getSettings().getMessageChannel(messageType.getKey()).equalsIgnoreCase("none")) {
+            return;
+        }
+
+        final DiscordMessageEvent event = new DiscordMessageEvent(messageType, FormatUtil.stripFormat(message), allowPing, avatarUrl, name, uuid);
+
+        // If the server is stopping, we cannot dispatch events.
+        if (messageType == MessageType.DefaultTypes.SERVER_STOP) {
+            jda.sendMessage(event, event.getMessage(), event.isAllowGroupMentions());
```

Comment:
Return after this? No point trying to post the event

Reply:
whoops